test/distributed/_tensor/test_dtensor.py

```diff
@@ -820,6 +820,53 @@ def test_split_tensor_1D(self) -> None:
                 ]
                 assert_array_equal(expected_is_tensor_empty, is_tensor_empty)
 
+    @with_comms
+    def test_weight_tying(self):
```

Comment:
I think we should add this test case to `tensor/parallel` instead? test_dtensor.py suppose to test some basic functionalities of DTensor that does not involve higher level TP API, but looks like we are using TP API, so let's put this to the TP test folder

Reply:
Yes I agree. I put it again into `test_tp_examples.py`. Please let me know if you think we'd better create a new test file for this.